inspector/plottools/doDockerPyplot

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+# doDockerPyplot [build] [shell]
+
+set -e
+DOCKER_TAG=measurementlab/nbrunner
+
+# deduce the user credentials: fail if not exactly one
+shopt -s nullglob
+CREDS=`echo ~/.config/gcloud/legacy_credentials/*`
+if [ `echo "$CREDS" | wc -w` != 1 ]; then
+  echo "Did not find exactly one credential: $CREDS"
+  exit 2
+fi
+
+if [ x$1 = xbuild ]; then
+  shift
+  sudo docker build . -t $DOCKER_TAG
+  echo "Consider: docker push $DOCKER_TAG"
+fi
+
+if [ x$1 = xshell ]; then
+  shift
+  doshell="-it --entrypoint=/bin/bash"
+fi
+
+DATADIR=$HOME/data
+
+set -x
+sudo docker run --net=host -v `pwd`:`pwd` -v $DATADIR:/data \
+  -v  ~/.config/gcloud/legacy_credentials/mattmathis@google.com:/creds \
+  -w `pwd` -e GOOGLE_APPLICATION_CREDENTIALS=/creds/adc.json \
+  --user $UID --group-add users \
+  $doshell $DOCKER_TAG
```

Comment:
Don't use the "all variables are implicitly defined to be """ trick.  If you want a shell variable to default to be the empty string, set it to the empty string.

Also, use `shellcheck` to lint your script.

Reply:
It seems that standard idioms from the past have become forbidden.  And a linter for shell?  When did that happen?

The above expansion of $doshell depends on word splitting - it is only set to constants elsewhere.